src/liboslexec/pointcloud.cpp

```diff
@@ -366,8 +369,24 @@ RendererServices::pointcloud_get (ShaderGlobals *sg,
     // then copy them back to the caller's indices.
 
     // Actual data query
-    const_cast<Partio::ParticlesData *>(cloud)->data (*attr, count, (Partio::ParticleIndex *)indices,
-                 true, out_data);
+    if (partio_type == OIIO::TypeString) {
+        // strings are special cases because they are stored as int index
+        std::unique_ptr<int[]> strindices(new int[count]);
```

Comment:
We have to malloc storage for all indices on every lookup?

Reply:
Well, let me take another pass at that. I guess we can do it only if there are more than one, which will be unusual. 

I really need to start using a good short vector implementation.

Reply:
Pushed an update

Reply:
Are you sure? I think in this use case there could be many. `count` is proportional to the number of points you found within your radius, right ?

Can we use `alloca` instead? (on the assumption that `count` is a reasonable value?)

Reply:
Yes, you are correct. I was misinterpreting as the number of attributes, rather than the number of points. My bad.
Will fix...

Reply:
Clearly I did not have enough coffee today. Update pushed.